DirectProgramming/C++SYCL_FPGA/Tutorials/Tools/use_library/part1_native/src/use_native.cpp

```diff
@@ -0,0 +1,107 @@
+//==============================================================
+// Copyright Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+#include <sycl/ext/intel/ac_types/ac_int.hpp>
+// oneAPI headers
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/sycl.hpp>
+
+#include "exception_handler.hpp"
+
+// Forward declare the kernel name in the global scope.
+// This FPGA best practice reduces name mangling in the optimization report.
+class KernelCompute;
+
+using MyInt27 = ac_int<27, false>;
+using MyInt54 = ac_int<54, false>;
+
+// Using host pipes to stream data in and out of kernel
+// IDPipeA and IDPipeB will be written to by the host, and then read by the kernel (device)
+// IDPipeC will be written to by the kernel (device), and then read by the host
+class IDPipeA;
+using InputPipeA = sycl::ext::intel::experimental::pipe<IDPipeA, unsigned>;
+class IDPipeB;
+using InputPipeB = sycl::ext::intel::experimental::pipe<IDPipeB, unsigned>;
+class IDPipeC;
+using OutputPipeC = sycl::ext::intel::experimental::pipe<IDPipeC, unsigned long>;
```

Comment:
replace unsigned long with uint64_t
replace unsigned with uint32_t
```suggestion
class IDPipeA;
using InputPipeA = sycl::ext::intel::experimental::pipe<IDPipeA, uint32_t>;
class IDPipeB;
using InputPipeB = sycl::ext::intel::experimental::pipe<IDPipeB, uint32_t>;
class IDPipeC;
using OutputPipeC = sycl::ext::intel::experimental::pipe<IDPipeC, uint64_t>;
```